File: src-tauri/src/protocol/topics.rs
pub fn control_topic(client_id: &str, session_id: &str) -> String {
    format!("meetings/{client_id}/session/{session_id}/control")
}

pub fn control_reply_topic(client_id: &str, session_id: &str) -> String {
    format!("meetings/{client_id}/session/{session_id}/control/reply")
}

pub fn events_topic(client_id: &str, session_id: &str) -> String {
    format!("meetings/{client_id}/session/{session_id}/events")
}

pub fn stt_topic(client_id: &str, session_id: &str) -> String {
    format!("meetings/{client_id}/session/{session_id}/stt")
}

pub fn summary_topic(client_id: &str, session_id: &str) -> String {
    format!("meetings/{client_id}/session/{session_id}/summary")
}

pub fn action_items_topic(client_id: &str, session_id: &str) -> String {
    format!("meetings/{client_id}/session/{session_id}/action-items")
}

#[cfg(test)]
mod tests {
    use super::{
        action_items_topic, control_reply_topic, control_topic, events_topic, stt_topic,
        summary_topic,
    };

    #[test]
    fn protocol_topics_build_expected_paths() {
        assert_eq!(
            control_topic("client-a", "session-1"),
            "meetings/client-a/session/session-1/control"
        );
        assert_eq!(
            control_reply_topic("client-a", "session-1"),
            "meetings/client-a/session/session-1/control/reply"
        );
        assert_eq!(
            events_topic("client-a", "session-1"),
            "meetings/client-a/session/session-1/events"
        );
        assert_eq!(
            stt_topic("client-a", "session-1"),
            "meetings/client-a/session/session-1/stt"
        );
        assert_eq!(
            summary_topic("client-a", "session-1"),
            "meetings/client-a/session/session-1/summary"
        );
        assert_eq!(
            action_items_topic("client-a", "session-1"),
            "meetings/client-a/session/session-1/action-items"
        );
    }
}

File: src/features/session/topics.ts
export function controlTopic(clientId: string, sessionId: string): string {
  return `meetings/${clientId}/session/${sessionId}/control`;
}

export function controlReplyTopic(clientId: string, sessionId: string): string {
  return `meetings/${clientId}/session/${sessionId}/control/reply`;
}

export function eventsTopic(clientId: string, sessionId: string): string {
  return `meetings/${clientId}/session/${sessionId}/events`;
}

export function sttTopic(clientId: string, sessionId: string): string {
  return `meetings/${clientId}/session/${sessionId}/stt`;
}

export function summaryTopic(clientId: string, sessionId: string): string {
  return `meetings/${clientId}/session/${sessionId}/summary`;
}

export function actionItemsTopic(clientId: string, sessionId: string): string {
  return `meetings/${clientId}/session/${sessionId}/action-items`;
}

